pkg/build/builder/docker.go

```diff
@@ -127,32 +136,48 @@ func (d *DockerBuilder) Build() error {
 				dockercfg.PullAuthType,
 			)
 			glog.V(0).Infof("\nPulling image %s ...", imageName)
+			startTime := unversioned.Now()
 			if err = pullImage(d.dockerClient, imageName, pullAuthConfig); err != nil {
 				d.build.Status.Phase = api.BuildPhaseFailed
 				d.build.Status.Reason = api.StatusReasonPullBuilderImageFailed
 				d.build.Status.Message = api.StatusMessagePullBuilderImageFailed
 				handleBuildStatusUpdate(d.build, d.client, nil)
 				return fmt.Errorf("failed to pull image: %v", err)
 			}
+			if d.build.Status.Stages, err = api.RecordStageAndStepInfo(d.build.Status.Stages, api.StagePullImages, api.StepPullBaseImage, startTime, unversioned.Now()); err != nil {
```

Comment:
you've got the context object, why not consistently call timing.RecordNewStep everywhere?  It's a bit confusing having two mechanisms for recording new steps being used in different places.